src/sqlfluff/core/linter/linter.py

```diff
@@ -0,0 +1,760 @@
+                # Handle potential .sqlfluffignore files
+                if ignore_files and fname == ignore_file_name:
+                    with open(fpath, "r") as fh:
+                        spec = pathspec.PathSpec.from_lines("gitwildmatch", fh)
+                    matches = spec.match_tree(dirpath)
+                    for m in matches:
+                        ignore_path = os.path.join(dirpath, m)
+                        ignore_set.add(os.path.abspath(ignore_path))
+                    # We don't need to process the ignore file any futher
+                    continue
+
+                # We won't purge files *here* because there's an edge case
+                # that the ignore file is processed after the sql file.
+
+                # Scan for remaining files
+                for ext in self.config.get("sql_file_exts", default=".sql").split(","):
+                    # is it a sql file?
+                    if fname.endswith(ext):
+                        buffer.append(fpath)
+
+        if not ignore_files:
+            return sorted(buffer)
+
+        # Check the buffer for ignore items and normalise the rest.
+        filtered_buffer = []
+
+        for fpath in buffer:
+            if os.path.abspath(fpath) not in ignore_set:
+                filtered_buffer.append(os.path.normpath(fpath))
+            elif is_exact_file:
+                linter_logger.warning(
+                    "Exact file path %s was given but "
+                    "it was ignored by a %s pattern, "
+                    "re-run with `--disregard-sqlfluffignores` to "
+                    "skip %s"
+                    % (
+                        path,
+                        ignore_file_name,
+                        ignore_file_name,
+                    )
+                )
+
+        # Return
+        return sorted(filtered_buffer)
+
+    def lint_string_wrapped(
+        self, string: str, fname: str = "<string input>", fix: bool = False
+    ) -> LintingResult:
+        """Lint strings directly."""
+        result = LintingResult()
+        linted_path = LintedPath(fname)
+        linted_path.add(self.lint_string(string, fname=fname, fix=fix))
+        result.add(linted_path)
+        return result
+
+    MIN_THRESHOLD_PARALLEL = 2
+    PARALLEL_CLS = runner_module.MultiProcessRunner
+
+    def lint_path(
+        self,
+        path: str,
+        fix: bool = False,
+        ignore_non_existent_files: bool = False,
+        ignore_files: bool = True,
+        parallel: int = 1,
+    ) -> LintedPath:
+        """Lint a path."""
+        linted_path = LintedPath(path)
+        if self.formatter:
+            self.formatter.dispatch_path(path)
+        fnames = list(
+            self.paths_from_path(
+                path,
+                ignore_non_existent_files=ignore_non_existent_files,
+                ignore_files=ignore_files,
+            )
+        )
+        runner: runner_module.BaseRunner

```

Comment:
After I recently wrote this section of code that selects the runner type, I wondered if I should move it to the file that implements the runner classes. I think that'd be a little tidier. Thoughts?

Reply:
Yes definitely. I was planning that as part of the second round of this PR. I might come back to this next rather than addressing it in this PR.

Reply:
No problem. I'm really pushing the limits, asking you to clean up my messy code. 😁